kobo/apps/superuser_stats/views.py

```diff
@@ -22,40 +31,227 @@ def _base_filename_to_full_filename(base_filename, username):
+    submission range
+    """
+    today = timezone.now().date()
+    base_filename = 'form-count-by-submissions-count-report_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        today,
+        datetime.now().microsecond
+    )
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username
+    )
+    generate_forms_count_by_submission_range.delay(filename)
+    template_ish = (
+        f'<html><head><title>Forms count by submissions count report.</title></head>'
+        f'<body>Your report is being generated. Once finished, it will be '
+        f'available at <a href="{base_filename}">{base_filename}</a>.<br>'
+        f'If you receive a 404, please refresh your browser periodically until '
+        f'your request succeeds.'
+        f'</body></html>'
+    )
+    return HttpResponse(template_ish)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def media_storage(request):
+    """
+    Generates a report which totals the amount of GB a user
+    has stored
+    """
+    base_filename = 'media_storage_report_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        date.today(),
+        datetime.now().microsecond
+    )
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username)
+    generate_media_storage_report.delay(filename)
     template_ish = (
-        '<html><head><title>Hello, superuser.</title></head>'
+        '<html><head><title>Media storage report</title></head>'
         '<body>Your report is being generated. Once finished, it will be '
-        'available at <a href="{0}">{0}</a>. If you receive a 404, please '
-        'refresh your browser periodically until your request succeeds.<br><br>'
-        'To select a date range, add a ? at the end of the URL and set the '
-        'start_date parameter to YYYY-MM-DD and/or the end_date parameter to '
-        'YYYY-MM-DD. Example:<br>'
-        'https://{{ kpi_base_url }}/superuser_stats/country_report/?start_date'
-        '=2020-01-31&end_date=2021-02-28'
+        f'available at <a href="{base_filename}">{base_filename}</a>.<br>'
+        'If you receive a 404, please refresh your browser periodically until '
+        'your request succeeds.'
         '</body></html>'
     ).format(base_filename)
+    return HttpResponse(template_ish)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def user_count_by_organization(request):
+    """
+    Generates a report that counts the number of users per organization
+    """
+    # Generate the file basename
+    base_filename = 'user-count-by-organization_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        date.today(),
+        datetime.now().microsecond
+    )
+
+    # Generate the CSV file
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username)
+    generate_user_count_by_organization.delay(filename)
```

Comment:
Maybe it would make sense to have the unspecified orgs have a name like "unknown" or "unspecified" rather than blank (there are also more users that haven't been included in those stats):


Organization | Count
-- | --
  | 5
Kobo | 1
KoBoToolbox | 2
test | 1
Tino's Super Org | 1

Reply:
Fixed!